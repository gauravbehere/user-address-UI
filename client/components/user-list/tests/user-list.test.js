import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import UserList from '..';

describe('Test suite for UserList component', () => {
  const mockUserList = [{name:'Abc', empId: 123, gender: 'M', id: 2}];

  it('Should render UserList component without props', () => {
    const { container, getByTestId } = render(<UserList/>);
    expect(container).toMatchSnapshot();
    expect(getByTestId('noUserMsg')).toBeInTheDocument();
  });

  it('Should render UserList component with error shown if there was an error fetching users', () => {
    const { getByText } = render(<UserList userListError='Error'/>);
    expect(getByText('Error')).toBeInTheDocument();
  });

  it('Should render UserList with the users passed', () => {
    const { getByText } = render(<UserList userList={mockUserList}/>);
    expect(getByText('Abc')).toBeInTheDocument();
  });

  it('Should let the parent component know about the user selected', () => {
    const onUserClicked = jest.fn();
    const { getByText, getByTestId } = render(<UserList userList={mockUserList} onUserClicked={onUserClicked}/>);
    expect(getByText('Abc')).toBeInTheDocument();
    fireEvent.click(getByTestId('user-list-item-0'));
    expect(onUserClicked).toHaveBeenCalledWith(2);
  });

  it('Should highlight the user selected', () => {
    const onUserClicked = jest.fn();
    const { getByTestId } = render(<UserList userList={mockUserList} onUserClicked={onUserClicked} selelectedUserId={2}/>);
    expect(getByTestId('user-list-item-0-active')).toBeInTheDocument();
  });
});
