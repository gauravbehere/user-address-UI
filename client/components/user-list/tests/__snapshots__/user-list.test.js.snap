// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for UserList component Should render UserList component without props 1`] = `
<div>
  <div
    class="user-list"
  >
    <div
      class="user-list-header"
    >
      List of Users
    </div>
    <div
      data-testid="noUserMsg"
    >
      No user found
    </div>
  </div>
</div>
`;
